### PR TITLE
Kvmtool

### PR DESCRIPTION
**Pull request by uudiin**
Introduce kvm backend support for skeleton

### CODE REVIEW

File: rune/libenclave/internal/runtime/pal/loader.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import "C"
 
 import (
 	"unsafe"
+	"github.com/sirupsen/logrus"
```
Comment (hustliyilin) [inline rune/libenclave/internal/runtime/pal/loader.go:13]: use `go fmt` before commiting your codes and you can see "github.com/sirupsen/logrus" is above "unsafe"

File: rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c [hunk 2]
```diff
@@ -576,6 +610,9 @@ int __pal_create_process(pal_create_process_args *args)
 		return -1;
 	}
 
+	if (backend_kvm)
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:613]: Is it possible to load an application here?

Comment (jiazhang0) [reply rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:613]: e.g, set the init process through kernel command line.

File: rune/libenclave/internal/runtime/pal/loader.go (hunk 1 above)
Comment (uudiin) [reply rune/libenclave/internal/runtime/pal/loader.go:12]: Formatted already.

File: rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c (hunk 2 above)
Comment (uudiin) [reply rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:618]: At present, the init process to be run can be specified by `runtime.args`, which is used to replace the default `/sbin/init`.

**Review by hustliyilin — APPROVED** [review]

**Review by jiazhang0 — APPROVED** [review]

**Review by haosanzi — APPROVED** [review]

File: rune/libenclave/internal/runtime/pal/loader.go [hunk 3]
```diff
@@ -23,7 +24,8 @@ var fptr_pal_get_local_report unsafe.Pointer
 func Loadbinary(path string) {
 	dl := C.dlopen(C.CString(path), C.RTLD_NOW)
 	if dl == nil {
-		C.perror(C.CString("failed to load library " + path))
+		logrus.Fatal("failed to load %s, dlerror: %s", path, C.GoString(C.dlerror()))
+		return
```
Comment (YangLiang3) [inline rune/libenclave/internal/runtime/pal/loader.go:28]: logrus.Fatal will call os.Exit(1), so return is not neccessary. Please remove return.

Comment (uudiin) [reply rune/libenclave/internal/runtime/pal/loader.go]: removed already. Thanks for pointing it out.

**Pull request merged**